projects/angular-auth-oidc-client/src/lib/utils/url/url.service.ts

```diff
@@ -40,7 +40,47 @@ export class UrlService {
     return results === null ? '' : decodeURIComponent(results[1]);
   }
 
-  isCallbackFromSts(currentUrl: string): boolean {
+  getUrlWithoutQueryParameters(url: URL): URL {
+    const u = new URL(url.toString());
+
+    const keys = [];
+
+    for (const key of u.searchParams.keys()) {
+      keys.push(key);
+    }
+
+    keys.forEach((key) => {
+      u.searchParams.delete(key);
+    });
+
+    return u;
+  }
+
+  queryParametersExist(expected: URLSearchParams, actual: URLSearchParams): boolean {
+    let r = true;
+
+    expected.forEach((v, k) => {
```

Comment:
Can we work with a `some` instead of a `foreach` and a variable? Have not tried it out.

Reply:
The `URLSearchParams` interface does not seem to have a `some`-method